[native] Update docs for VARCHAR type length value (#24173)
prestodb/presto: presto-docs/src/main/sphinx/language/types.rst

.. note::

>     For compatibility reasons decimal literals without explicit type specifier (e.g. ``1.2``)

Reviewer: ```suggestion
    For compatibility reasons decimal literals without explicit type specifier such as ``1.2``
```

>     are treated as values of the ``DOUBLE`` type by default up to version 0.198.

Reviewer 2: Just specify the current behavior. It doesn't matter what happened many versions ago.
